File: frontend/src/types/task.ts
export const TASK_STATUSES = ["pending", "completed"] as const;

export type TaskStatus = (typeof TASK_STATUSES)[number];

export type TaskFilter = TaskStatus | "all";

export interface Task {
  id: number;
  title: string;
  description: string | null;
  status: TaskStatus;
  creation_date: string;
}

export interface CreateTaskPayload {
  title: string;
  description?: string | null;
  status: TaskStatus;
}

export interface UpdateTaskPayload {
  title?: string;
  description?: string | null;
  status?: TaskStatus;
}

export interface TaskFormValues {
  title: string;
  description: string;
  status: TaskStatus;
}

export interface TaskFormErrors {
  title?: string;
  status?: string;
}

File: app/models/task.py
from __future__ import annotations

"""Task ORM model."""

from datetime import datetime
from enum import Enum
from typing import Optional

from sqlalchemy import DateTime, Enum as SqlEnum, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class TaskStatus(str, Enum):
    """Allowed task statuses."""

    PENDING = "pending"
    COMPLETED = "completed"


class Task(Base):
    """Task database entity."""

    __tablename__ = "tasks"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    title: Mapped[str] = mapped_column(String(120), nullable=False)
    description: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    status: Mapped[TaskStatus] = mapped_column(
        SqlEnum(TaskStatus, name="task_status"),
        default=TaskStatus.PENDING,
        nullable=False,
    )
    creation_date: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )
